phoenix-core/src/main/java/org/apache/phoenix/expression/LiteralExpression.java

```diff
@@ -57,181 +56,236 @@
 	private static final LiteralExpression[] NULL_EXPRESSIONS = new LiteralExpression[Determinism.values().length];
     private static final LiteralExpression[] TYPED_NULL_EXPRESSIONS = new LiteralExpression[PDataType.values().length * Determinism.values().length];
     private static final LiteralExpression[] BOOLEAN_EXPRESSIONS = new LiteralExpression[2 * Determinism.values().length];
-    
+
     static {
     	for (Determinism determinism : Determinism.values()) {
-    		NULL_EXPRESSIONS[determinism.ordinal()] = new LiteralExpression(null, determinism);
+            NULL_EXPRESSIONS[determinism.ordinal()] = new LiteralExpression(null, null,
+                    ByteUtil.EMPTY_BYTE_ARRAY, null, null, SortOrder.getDefault(), determinism);
 	        for (int i = 0; i < PDataType.values().length; i++) {
-	            TYPED_NULL_EXPRESSIONS[i+PDataType.values().length*determinism.ordinal()] = new LiteralExpression(PDataType.values()[i], determinism);
-	        }        
-	        BOOLEAN_EXPRESSIONS[determinism.ordinal()] = new LiteralExpression(Boolean.FALSE,
-              PBoolean.INSTANCE, PBoolean.INSTANCE.toBytes(Boolean.FALSE), determinism);
-	        BOOLEAN_EXPRESSIONS[Determinism.values().length+determinism.ordinal()] = new LiteralExpression(Boolean.TRUE, PBoolean.INSTANCE, PBoolean.INSTANCE.toBytes(Boolean.TRUE), determinism);
+	            PDataType type = PDataType.values()[i];
+                TYPED_NULL_EXPRESSIONS[i+PDataType.values().length*determinism.ordinal()] =
+                        new LiteralExpression(null, type, ByteUtil.EMPTY_BYTE_ARRAY,
+                                !type.isFixedWidth() ? null : type.getMaxLength(null), null,
+                                SortOrder.getDefault(), determinism);
+	        }
+            BOOLEAN_EXPRESSIONS[determinism.ordinal()] = new LiteralExpression(Boolean.FALSE,
```

Comment:
Using a Builder here has the potential of throwing an SQL Exception when the build method attempts to derive the `value` field, which isn't necessary since the value has already been provided.